integer/src/chip/mul.rs

```diff
@@ -167,12 +168,12 @@ impl<W: FieldExt, N: FieldExt, const NUMBER_OF_LIMBS: usize, const BIT_LEN_LIMB:
         }
 
         // Constrain binary part of crt
-        self.constrain_binary_crt(ctx, &t.try_into().unwrap(), result, residues)?;
+        self.constrain_binary_crt(ctx, &t.try_into().unwrap(), &result, residues)?;
```

Comment:
Maybe use `expect` instead? With a message referring to the conversion that failed.